### ISSUE TITLE
Inline hook handling of relative jumps not proper?

### ISSUE THREAD

**Issue by gibbed** [cursey/safetyhook#86]
https://github.com/cursey/safetyhook/blob/de83823750b604c83b93bc80683b02978d3ae589/src/inline_hook.cpp#L260-L284

I'm creating a `MidHook` at the address `00000001004751AC`, which has the following instructions:

```
00000001004751AC  72 AE              jb  short 10047515C
00000001004751AE  48 8D BD E0F8FFFF  lea rdi, [rbp-720h]
```

this ends up emitting the trampoline:

```
00000000FFFF0327  0F82 AEFFFFFF      jb  FFFF02DB
00000000FFFF032D  48 8D BD E0F8FFFF  lea rdi, [rbp-720h]
00000000FFFF0334  E9 7C4E4800        jmp 1004751B5
```

The `jb` here did not get resolved properly (it should be a relative jmp to `000000010047515C`).

`target_address < m_target + m_original_bytes.size()`

ends up being

`000000010047515C < 00000001004751B5`

So I'm guessing that check should actually be:

`target_address >= m_target && target_address < m_target + m_original_bytes.size()`

**Comment by praydog** [comment]
Sounds like a reasonable fix to me.

**Comment by cursey** [comment]
@gibbed can you try #87 and let me know if it actually fixes the issue you've described?

**Comment by gibbed** [comment]
Yeah, it does. Though you've missed the check for unconditional jumps.

https://github.com/cursey/safetyhook/pull/87/files#diff-b109456f39b67560bebff92603a5d7152ce3a5188182bc84c231e6c3e7019f7cL278
